language-server/test/dotty/tools/languageserver/SignatureHelpTest.scala

```diff
@@ -17,12 +17,52 @@ class SignatureHelpTest {
       .signatureHelp(m1, List(signature), Some(0), 0)
   }
 
+  @Test def properFunctionReturnWithoutParenthesis: Unit = {
+    val listSignature = S("apply", List("A"), List(List(P("elems", "A*"))), Some("List[A]"))
+    val optionSignature = S("apply", List("A"), List(List(P("x", "A"))), Some("Option[A]"))
+    code"""object O {
+          |  List(1, 2$m1
+          |}
+          |object T {
+          |  List(Option(1$m2
+          |}
+          |object Z {
+          |  List(Option(1)$m3
+          |}"""
+      .signatureHelp(m1, List(listSignature), Some(0), 0)
+      .signatureHelp(m2, List(optionSignature), Some(0), 0)
+      .signatureHelp(m3, List(listSignature), Some(0), 0)
+  }
+
+  @Test def partialyFailedCurriedFunctions: Unit = {
+    val listSignature = S("curry", Nil, List(List(P("a", "Int"), P("b", "Int")), List(P("c", "Int"))), Some("Int"))
+    code"""object O {
+          |def curry(a: Int, b: Int)(c: Int) = a
+          |  curry(1$m1)(3$m2)
+          |}"""
+      .signatureHelp(m1, List(listSignature), Some(0), 0)
+      .signatureHelp(m2, List(listSignature), Some(0), 2)
+  }
+
+  @Test def optionProperSignature: Unit = {
+    val listSignature = S("apply", List("A"), List(List(P("x", "A"))), Some("Option[A]"))
```

Comment:
Super minot, but this mislead me a bit
```suggestion
    val optionSignature = S("apply", List("A"), List(List(P("x", "A"))), Some("Option[A]"))
```